src/app/chart/donut/donut.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, DoCheck, Input, OnInit } from '@angular/core';
+
+import { cloneDeep, defaults, isEqual, merge } from 'lodash';
+import { ChartDefaults } from '../chart.defaults';
+import { ChartBase } from '../chart.base';
+import { DonutConfig } from './donut-config';
+
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'pfng-chart-donut',
+  templateUrl: './donut.component.html'
+})
+export class DonutComponent extends ChartBase implements DoCheck, OnInit {
+
+  @Input() chartData: any;
+  @Input() config: DonutConfig;
+
+  public donutChartId: any;
+
+  private prevChartData: any;
+  private defaultConfig: DonutConfig;
+
+  /**
+   * Default constructor
+   * @param chartDefaults
+   */
+  constructor(private chartDefaults: ChartDefaults) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.donutChartId = 'donutChart';
```

Comment:
There could be more than one chart on a page, so assigning the same ID won't scale well. 

I realize this is a direct port of APF, but this looks like a bug to me. I'm thinking we append some sort of auto-generated number or just leave it blank?

If the chart needs an id to render, then we can throw an exception, forcing the dev to set one?

Reply:
Right. I like the idea of auto-generating a uuid to attach behind the scenes. When using the sparkline chart in f8-ui I ran into the chart id issue where id's on the same page need to be unique (whereas for some other things, the id's only need to be unique for the same angular template). So multiple templates on the same page could conflict if the id's happened to be the same for the charts. It would be nice to have an auto-generated uniqueness behind the scenes.